### ISSUE TITLE
error: constructor Builder in class Builder cannot be applied to given types;

### ISSUE THREAD

**Issue by Dineshshrinath** [phonegap/phonegap-plugin-push#2786]
I am getting error as below when I build the cordova application. For debugging I am using Android studio.

:app:compileDebugJavaWithJavacD:\MobileProfilerProjects\ADvisorMobileAPP\Liferay7.0-Plugin\advisor\platforms\android\app\src\main\java\com\adobe\phonegap\push\FCMService.java:393: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
D:\MobileProfilerProjects\ADvisorMobileAPP\Liferay7.0-Plugin\advisor\platforms\android\app\src\main\java\com\adobe\phonegap\push\FCMService.java:403: error: constructor Builder in class Builder cannot be applied to given types;
        mBuilder = new NotificationCompat.Builder(context, channelID);
                   ^
  required: Context
  found: Context,String
  reason: actual and formal argument lists differ in length
D:\MobileProfilerProjects\ADvisorMobileAPP\Liferay7.0-Plugin\advisor\platforms\android\app\src\main\java\com\adobe\phonegap\push\PushPlugin.java:308: error: cannot find symbol
                "has permission: " + NotificationManagerCompat.from(getApplicationContext()).areNotificationsEnabled());
                                                                                            ^
  symbol:   method areNotificationsEnabled()
  location: class NotificationManagerCompat
D:\MobileProfilerProjects\ADvisorMobileAPP\Liferay7.0-Plugin\advisor\platforms\android\app\src\main\java\com\adobe\phonegap\push\PushPlugin.java:309: error: cannot find symbol
            jo.put("isEnabled", NotificationManagerCompat.from(getApplicationContext()).areNotificationsEnabled());
                                                                                       ^
  symbol:   method areNotificationsEnabled()
  location: class NotificationManagerCompat
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 errors
 FAILED

In FCMService.java it is showing error at 
 mBuilder = new NotificationCompat.Builder(context, channelID);

if I check .Builder method then it shows me right constructor as, (Notificationcompact.java)
    public Builder(@NonNull Context context, @NonNull String channelId) {

Also for other error it says "error: cannot find symbol method areNotificationsEnabled()"
  Log.d(LOG_TAG,
                "has permission: " + NotificationManagerCompat.from(getApplicationContext()).areNotificationsEnabled());

but if I check  areNotificationsEnabled() method it takes me to areNotificationsEnabled() method in NotificationmanagerCompact.java,

I am not sure about which jar reference I am missing. Could you please help.

**Comment by yucomds** [comment]
I'm facing the same issue.  Did you solve the problem?

**Comment by Dineshshrinath** [comment]
Yes. It was due to another reference added to android support library. In my project there was android support library added as plugin. It was old one. so I removed that library and it worked. 
You can open your project in android studio and check the methods showing error. It should show the proper method version.
When you built it or compile it it refers the old library. Hence the error.

**Comment by gtcdevop** [comment]
@Dineshshrinath  how did you mange to solve, could you paste the solution here
